library/src/dotty/DottyPredef.scala

```diff
@@ -38,4 +38,32 @@ object DottyPredef {
   }
 
   inline def summon[T](given x: T): x.type = x
+
+  // Extension methods for working with explicit nulls
+
+  /** Strips away the nullability from a value.
+   *  e.g.
+   *    val s1: String|Null = "hello"
+   *    val s: String = s1.nn
+   *
+   *  Note that `.nn` performs a checked cast, so if invoked on a null value it'll throw an NPE.
+   */
+  def[T] (x: T|Null) nn: T =
+    if (x == null) throw new NullPointerException("tried to cast away nullability, but value is null")
+    else x.asInstanceOf[T]
+
+  /** Reference equality where the receiver is a nullable union.
+   *  Note that if the receiver `r` is a reference type (e.g. `String`), then `r.eq` will invoke the
+   *  `eq` method in `AnyRef`.
+   */
+  def (x: AnyRef|Null) eq(y: AnyRef|Null): Boolean =
+    (x == null && y == null) || (x != null && x.eq(y))
```

Comment:
As mentioned above, this can be more efficient as:
```suggestion
    x.asInstanceOf[AnyRef] eq y.asInstanceOf[AnyRef]
```

Reply:
This method no longer exists.